base_layer/wallet/src/utxo_scanner_service/utxo_scanner_task.rs

```diff
@@ -449,6 +456,36 @@ where
                 let response = response?;
                 #[allow(clippy::cast_possible_wrap)]
                 for response in response.blocks {
+                    if let Some(previous_block) = &prev_scanned_block {
+                        if response.height < previous_block.height {
+                            // We do not accept blocks that go backwards in height - fork block re-validation forced.
+                            return Err(anyhow!(
+                                "Non-monotonic block heights received during UTXO scan: {} followed by {}",
+                                previous_block.height,
+                                response.height
+                            ));
+                        }
+                        if response.height > previous_block.height + 1 {
+                            // Missing block(s) detected between previous_block and response height - fork block
+                            // re-validation forced.
+                            return Err(anyhow!(
+                                "Non-consecutive block heights received during UTXO scan: {} followed by {}",
+                                previous_block.height,
+                                response.height
+                            ));
+                        }
+                        if response.height == previous_block.height &&
```

Comment:
this is possible and happens. 
the base node will split the block into seperate blocks if the block is too big, more than 2000 inputs/outputs.

Reply:
Yes, we check here that if the height is the same, the hashes may not be different.